de.metas.business/src/main/java/org/adempiere/impexp/product/MProductImportTableSqlUpdater.java

```diff
@@ -414,7 +414,7 @@ private void dbUpdateTaxCategories(@NonNull final String whereClause, @NonNull f
 		{
 			sql = new StringBuilder("UPDATE ")
 					.append(targetTableName + " i ")
-					.append(" set C_TaxCategory_ID=(select tc.C_TaxCategory_ID from C_TaxCategory tc where tc.IsDefault='Y' and tc.AD_Client_ID=")
+					.append(" set C_TaxCategory_ID=(select tc.C_TaxCategory_ID from C_TaxCategory tc where tc.VATType='N' and tc.AD_Client_ID=")
```

Comment:
why not trying to fetch the C_TaxCategory_ID by using some API, and then here, just simple set it.
In that way we would prevent errors like the one u've fixed here.